management/WFCORE-7235_Platform_MBeans_Refresh.adoc

```diff
@@ -0,0 +1,182 @@
+  * `total-swap-space-size`: The total amount of swap space in bytes.
+
+=== Evolutions of the RuntimeMXBean
+
+==== New standard attributes
+
+  * `pid`: The process ID representing the running Java virtual machine.
+
+=== Evolutions of the ThreadMXBean
+
+==== New standard fields in `ThreadInfo`
+
+  * `daemon`: Whether the thread is a daemon thread.
+  * `priority`: The thread priority of the thread.
+
+==== New standard fields in `StackTraceElement`
+
+  * `class-loader-name`: The name of the class loader of the class containing the execution point represented by this stack trace element.
+  * `module-name`: The module name of the module containing the execution point represented by this stack trace element.
+  * `module-version`: The module version of the module containing the execution point represented by this stack trace element.
+
+==== New attributes from the `com.sun.management` package
+
+  * `current-thread-allocated-bytes`: An approximation of the total amount of memory, in bytes, allocated in heap memory for the current thread.
+  * `thread-allocated-memory-enabled`: Tests if thread memory allocation measurement is enabled.
+  * `thread-allocated-memory-supported`: Tests if the Java virtual machine implementation supports thread memory allocation measurement.
+
+==== New operations from the `com.sun.management` package
+
+  * `get-thread-allocated-bytes`: An approximation of the total amount of memory, in bytes, allocated in heap memory for the thread with the specified ID.
+  * `get-threads-allocated-bytes`: An approximation of the total amount of memory, in bytes, allocated in heap memory for each thread whose ID is in the input array ids.
+  * `get-thread-cpu-times`: The total CPU time for each thread whose ID is in the input array ids in nanoseconds.
+  * `get-thread-user-times`: The CPU time that each thread whose ID is in the input array ids has executed in user mode in nanoseconds.
+
+==== Implementation notes
+
+* The access to the `com.sun.management` MXBeans is reflective, WildFly can run on a JVM not defining them.
+* When reading a WildFly model attribute that does exist but is not supported by the JVM, the attribute value is set to "undefined".
+* When writing a WildFly model attribute that does exist but is not supported by the JVM, an OperationFailed exception is thrown.
+* When invoking a WildFly model operation that does exist but is not supported by the JVM, an OperationFailed exception is thrown.
+
+=== Hard Requirements
+
+* Expose attributes and operations that are defined in JDK11 API.
+* Expose the attributes and operations defined in MXBean in the `com.sun.management` package statically in the management model. 
+If the running JVM does not support them it means that attributes will be undefined and invoking operations will lead to an error.
+
```

Comment:
Please add something about the kernel management API version. It needs to be bumped, so describe precisely what you expect it to move to.

Reply:
Questions and doubts were resolved in Zulip, ref: https://wildfly.zulipchat.com/#narrow/channel/174184-wildfly-developers/topic/WFCORE-7235.20core-service.20and.20Stability.20levels/with/523926036

This convesation can be resolved